02-peer-protocol.md

```diff
@@ -445,6 +446,206 @@ to broadcast the commitment transaction to get his funds back and open a new
+they both send consecutive `tx_complete`.
+
+### Requirements
+
+(Note that many of these messages have their own, additional
+requirements detailed elsewhere).
+
+The initiator:
+- MUST `tx_add_input` an input which spends the current funding transaction output.
+- MUST `tx_add_output` a zero-value output which pays to the two funding keys using the higher of the two `generation` fields.
+- MUST pay for the common fields.
+
+Each node:
+- MUST pay for their own added inputs and outputs.
+
+Upon receipt of consecutive `tx_complete`s, each node:
+- MUST fail negotiation if there is not exactly one input spending the current funding transaction.
+- MUST fail negotiation if there is not exactly one output with zero value paying to the two funding keys (a.k.a. the new channel funding output)
+- MUST calculate the channel capacity for each side:
+  - Start with the previous balance
+  - Add that side's new inputs (excluding the one spending the current funding transaction)
+  - Subtracting each sides new outputs (except the zero-value one paying to the funding keys)
+  - Subtract the total fee that side is paying for the splice transaction.
+- MUST replace the zero-value funding output amount with the total channel capacity.
+- MUST calculate the channel balance for each side:
+  - Subtract any outstanding HTLCs offered by that side.
+- If either side has added an output other than the new channel funding output:
+  - MUST fail the negotiation if the balance for that side is less than 1% of the total channel capacity.
+- SHOULD NOT fail if the splice transaction is nonstandard.
+- MUST send a single `commitment_signed` for the splice transaction.
+  - Peer MUST NOT reply with `revoke_and_ack`
+
+- If recipient's sum(tx_add_input.amount) < peer's sum(tx_add_input.amount); or
+if recipient's sum(tx_add_input.amount) == peer's sum(tx_add_input.amount) and
+recipient is the `initiator` of the splice:
+  - SHOULD send `tx_signatures` first for the splice transaction.
+- else
+  - MAY send `tx_signatures` first.
+
+- Upon receipt of `tx_signatures` for the splice transaction:
+  - MUST consider splice negotiation complete.
+  - MUST consider the connection no longer quiescent.
+
+On reconnection:
+- MUST retransmit the last splice `tx_signatures` (if any).
+- MUST ignore any redundant `tx_signatures` it receives.
+
+### Rationale
+
+If a side does not meet the reserve requirements, that's OK: but if
+they take funds out of the channel, they must ensure that they do meet
+them.  If your peer adds a massive amount to the channel, then you
+only have to add more reserve if you want to contribute to the splice
+(and you can use `tx_remove_output` and/or `tx_remove_input` part-way
+through if this happens).
+
+## Splice Completion
+
+### The `splice_locked` Message
+
+1. type: 77 (`splice_locked`)
+2. data:
+   * [`channel_id`:`channel_id`]
+
+`channel_id` is the pre-splice channel's `channel_id`.
+
+Each node:
+- if any splice transaction reaches depth 6:
```

Comment:
I think really depth should be a `confirmation_depth` in the `splice` message. Some users might wish a lower or bigger amount, in function of the `funding_satoshis` committed in the channel and their liquidity prefs.

Reply:
I believe as structured just sending `splice_locked` early (say 1 conf) should be fine. If the other node requires 6, it will remember you sent it and send it back after it sees 6.

It's important to resend it on restart as it's not a state that is preserved across restarts.

This behavior is already needed to handle blocks coming in at mismatched times.

Reply:
we could make the depth of 6 a recommendation here

Reply:
> I believe as structured just sending splice_locked early (say 1 conf) should be fine. If the other node requires 6, it will remember you sent it and send it back after it sees 6.

The `confirmation_depth` is a security parameter which should be scaled in function of the funds at stake. A miner can always reorg a low-depth confirmed splicing transaction to double-spend a high-value HTLC (and note the batching of block reorg against unbounded number of channels).

Of course, you can send the `splice_locked` once the depth has reached your local node satisfying value. However you’re depriving your counterparty of visibility on when the channel is ready to route / forward HTLC.